### PR TITLE
Month picker rework

### PR DESCRIPTION
**Pull request by Miodec**
Reworked the budget month picker. Changes include:

 - Moved the month picker into its own file.
 - Variable amount of months displayed. The number of months displayed will now depend on the parent width. Years are displayed above the first visible month of that year.
image

 - Selected months preview on hover. The selected months will be highlighted to show the user the resulting month range.

https://github.com/actualbudget/actual/assets/13181393/d551d271-e5ff-4c3b-9cf1-ff430a10f70c

 - Selection stays in the center (this also elimiated the need for the arrow buttons)

https://github.com/actualbudget/actual/assets/13181393/8fc949b6-b8a3-4583-a01c-d67ae8900d7e

Other minor changes:
 - Changed the active theme indicator from underline to bold (as discussed on Discord with @j-f1)

Other notes:
 - I originally intended for the years different to the current one to be gray, but Jed said that it might be confusing to users (as it might appear that they are disabled)
image

### CODE REVIEW

**Comment by Shazib** [comment]
As an UW user, can there be some kind of cap on the months in focus? I'm not sure i'm really even interested in having a quick button to go to a random month two years in the past, so its just sort of noise?

image

**Comment by Miodec** [comment]
> As an UW user, can there be some kind of cap on the months on display? I'm not sure i'm really ever interested in having a quick button to go to a random month two years in the past, so its just sort of noise?
> 
> image

12 months into the past and future? Or would that still be too much you think?

**Comment by Shazib** [comment]
> > As an UW user, can there be some kind of cap on the months on display? I'm not sure i'm really ever interested in having a quick button to go to a random month two years in the past, so its just sort of noise?
> > image
> 
> 12 months into the past or future? Or would that still be too much you think?

I think that would be ok, can run it by the DC?

**Comment by Miodec** [comment]
image

Here's 24 month max

**Comment by MatissJanis** [comment]
Super cool! 

Yes, I think capping this at 24 makes sense. Also would be very cool if you could use keyboard arrows (<-,  ->) to navigate through the months (scope creep; don't feel pressured to actually do this).

Other than the capping - LGTM.

File: packages/desktop-client/src/components/budget/MonthPicker.js [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+                  textDecoration: 'line-through',
+                  color: colors.n7,
+                },
+                styles.smallText,
+                selected && {
+                  backgroundColor: colors.p6,
+                  color: 'white',
+                },
+                (hovered || selected) && {
+                  borderRadius: 0,
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/budget/MonthPicker.js:117]: **🥜 nitpick:** ‏would you mind adding a `cursor: pointer` hover state for the buttons?

**Review by MatissJanis — APPROVED** [review]
Very impressive work here! Thanks!

edit: oops, forgot about the capping at 24. Would you mind adding that in too?

**Comment by Miodec** [comment]
On holiday right now - will do in a few days

**Comment by Miodec** [comment]
On holiday right now - will do in a few days

**Comment by Miodec** [comment]
On holiday rn - will do in a few days

**Comment by Miodec** [comment]
Wait never mind, the 24 cap should already be on the PR

File: packages/desktop-client/src/components/budget/MonthPicker.js (hunk 1 above)
Comment (Miodec) [reply packages/desktop-client/src/components/budget/MonthPicker.js:117]: Done

**Comment by Miodec** [comment]
@MatissJanis updated the cursor

**Comment by Miodec** [comment]
Weird that adding cursor pointer would cause the build to fail? This can't be my fault right?

**Comment by j-f1** [comment]
I’m guessing the problem is that the tests fall out of sync with what is being built by Netlify. This has been kind of an annoying persistent problem, but updating the branch should fix it.

**Review by j-f1 — APPROVED** [review]

**Comment by j-f1** [comment]
Thank you for reworking this!

**Pull request merged**